src/Sylius/Bundle/ApiBundle/CommandHandler/Checkout/AddressOrderHandler.php

```diff
@@ -64,24 +70,24 @@ public function __invoke(AddressOrder $addressOrder): OrderInterface
         );
 
         /** @var CustomerInterface|null $customer */
-        $customer =  $order->getCustomer();
+        $customer = $this->customerRepository->findOneBy(['email' => $addressOrder->email]);
```

Comment:
I guess that only if there is no customer assigned to an order, we should search customer in the repository, because in case that customer is logged in, the email could be passed as null.